keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1256 @@
+
+There are also similar use cases for requesting the device based on a specific root or device type in other contexts,
+including [Peer Pods](https://github.com/confidential-containers/cloud-api-adaptor/blob/main/docs/architecture.md)
+and [Composable Disaggregated Infrastructure (CDI)](https://www.wwt.com/article/an-introduction-to-composable-disaggregated-infrastructure).
+
+This feature is also beneficial for the other sharable devices those are not with a scope of [KEP-4815](https://github.com/kubernetes/enhancements/issues/4815).
+For instance, this feature will be allow reserving memory fraction of virtual GPU in [the AWS virtual GPU device plugin](https://github.com/awslabs/aws-virtual-gpu-device-plugin) via DRA.
+The number of shares can be limited if needed.
+
+### Goals
+
+- Introduce an ability to allocating on-demand shared devices via DRA. This should cover the use cases of macvlan or ipvlan in a DRA driver for CNI and virtual accelerator devices with on-demand memory fraction.
+- Enhance a capability of secondary networks to dynamically allocate secondary networks based on present availabilities such as bandwidth.
+- Leverage the enhancement of capacity consumption from [KEP-4815](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4815-dra-partitionable-devices).
+
+### Non-Goals
+
+- Define driver-specific attributes and configs (such as CNI parameter config).
+- Support network security policy.
+- Add a flag to allow sharing device within the same pod (has no use case).
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+#### Story 1
+A user requests a secondary network based on their bandwidth demands without wanting to specify a device name nor present available bandwidth of each specific device. The DRA scheduler can dynamically select available devices based on their availability.
+
+```yaml
+kind: ResourceSlice
+...
+spec:
+  driver: cni.dra.networking.x-k8s.io
+  devices:
+  - name: eth1
+    basic:
+      attributes:
+        name:
+          string: "eth1"
+      consumableCapacity:
+        bandwidth:
+          value: 10Gi
+---
+kind: DeviceClass
+metadata:
+  name: vlan-cni.networking.x-k8s.io
+```
+
+```yaml
+kind: ResourceClaim
+...
+spec:
+  devices:
+    requests:
+    - name: macvlan
+      deviceClassName: vlan-cni.networking.x-k8s.io
+      allocationMode: Shared
+      resources:
+        requests:
+          bandwidth: "1Gi"
```

Comment:
I do not understand how is this going to be related to eth1, what if there is eth2 and eth3 too?

Reply:
User can specify the device via the selection (such as specify device's attribute name = eth1). Additionally, DRA allows pod to dynamically allocate device based on its available resource such as bandwidth.The selected device information will be also sent to the driver then the network driver and configure the selected device.

Reply:
I assume the idea here is to underspecify when possible, but be more specific if needed. So if there are also `eth2` and `eth3`, then leaving it like this means that any one of `eth1-3` will work. If that is not the case, then a selector is needed to further narrow down the eligible devices.

Reply:
Yes exactly, Morten. We can "schedule" onto any of the underlying eth* devices. If we want a specific one - or one with a specific attribute (like network or trunking) - then that should be specified like any other selector.